### PR TITLE
feat: add more consensus metrics on rejected proposals

### PR DESCRIPTION
**Pull request by cmwaters**
## Description

This adds two counters to the consensus metrics:

- ApplicationRejectedProposals
- TimedOutProposals

It also adds a third `StartHeight` which is set once upon start so we know how many heights the metrics have been running for

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use
  [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

**Review by rootulp — APPROVED** [review]
Do we want to backport v0.34.x-celestia?

**Comment by cmwaters** [comment]
> Do we want to backport v0.34.x-celestia?

Yup

File: consensus/state.go [hunk 1]
```diff
@@ -1297,12 +1300,14 @@ func (cs *State) defaultDoPrevote(height int64, round int32) {
 	stateMachineValidBlock, err := cs.blockExec.ProcessProposal(cs.ProposalBlock, cs.state)
 	if err != nil {
 		cs.Logger.Error("state machine returned an error when trying to process proposal block", "err", err)
+		return
```
Comment (staheri14) [inline consensus/state.go:1303]: [Question] Is the addition of this `return` statement intentional?

**Review by staheri14 — APPROVED** [review]

File: consensus/state.go (hunk 1 above)
Comment (cmwaters) [reply consensus/state.go:1303]: Yes, I don't think we should vote if there is an error here

**Pull request merged**